File: Data Structures and Algorithms/4. Recursion/Kunal/Pattern/P1.java
/*
******
******
******
******
*/

import java.util.*;
class P1 {
    public static void Rectangle(int rows, int cols, int N) {
        if(N == rows)
            return;
        if(cols < rows) {
            System.out.print("* ");
            Rectangle(rows, cols + 1, N);
        }
        else {
            System.out.println();
            Rectangle(rows, 0, N + 1);
        }
    }

    public static void main(String[] args) {
        Rectangle(4, 0, 0);
    }
}

File: Data Structures and Algorithms/4. Recursion/Kunal/Pattern/p1.cpp
#include <iostream>
using namespace std;

void rectangle(int rows, int cols, int N) {
    if(N == 0)
        return;
    if(cols < rows){
        cout << "* ";
        rectangle(rows, cols + 1, N);
    }
    else {
        cout << "\n";
        rectangle(rows, 0, N - 1);
    }
}
 
int main() {
    rectangle(4, 0, 4);
    return 0;
}
